model/file_info_test.go

```diff
@@ -24,109 +24,186 @@ func TestFileInfoIsValid(t *testing.T) {
+			expectedWidth:           408,
+			expectedHeight:          336,
+			expectedHasPreviewImage: true,
+		},
+		{
+			testName:                "Static Gif File",
+			filename:                "handtinywhite.gif",
+			file:                    gifFile,
+			usePrefixForMime:        false,
+			expectedExtension:       "gif",
+			expectedSize:            35,
+			expectedMime:            "image/gif",
+			expectedWidth:           1,
+			expectedHeight:          1,
+			expectedHasPreviewImage: true,
+		},
+		{
+			testName:                "Animated Gif File",
+			filename:                "testgif.gif",
+			file:                    animatedGifFile,
+			usePrefixForMime:        false,
+			expectedExtension:       "gif",
+			expectedSize:            38689,
+			expectedMime:            "image/gif",
+			expectedWidth:           118,
+			expectedHeight:          118,
+			expectedHasPreviewImage: false,
+		},
+		{
+			testName:                "No extension File",
+			filename:                "filewithoutextension",
+			file:                    fakeFile,
+			usePrefixForMime:        false,
+			expectedExtension:       "",
+			expectedSize:            1000,
+			expectedMime:            "",
+			expectedWidth:           0,
+			expectedHeight:          0,
+			expectedHasPreviewImage: false,
+		},
+		{
+			// Always make the extension lower case to make it easier to use in other places
+			testName:                "Uppercase extension File",
+			filename:                "file.TXT",
+			file:                    fakeFile,
+			usePrefixForMime:        true,
+			expectedExtension:       "txt",
+			expectedSize:            1000,
+			expectedMime:            "text/plain",
+			expectedWidth:           0,
+			expectedHeight:          0,
+			expectedHasPreviewImage: false,
+		},
+		{
+			// Don't error out for image formats we don't support
+			testName:                "Not supported File",
+			filename:                "file.tif",
+			file:                    fakeFile,
+			usePrefixForMime:        false,
+			expectedExtension:       "tif",
+			expectedSize:            1000,
+			expectedMime:            "image/tiff",
+			expectedWidth:           0,
+			expectedHeight:          0,
+			expectedHasPreviewImage: false,
+		},
+	}
+
+	for _, tc := range ttc {
+		t.Run(tc.testName, func(t *testing.T) {
+			info, errApp := GetInfoForBytes(tc.filename, tc.file)
+			require.Nil(t, errApp)
+
+			assert.Equalf(t, info.Name, tc.filename, "Got incorrect filename: %v", info.Name)
+			assert.Equalf(t, info.Extension, tc.expectedExtension, "Got incorrect extension: %v", info.Extension)
+			assert.EqualValuesf(t, info.Size, tc.expectedSize, "Got incorrect size: %v", info.Size)
+			assert.Equalf(t, info.Width, tc.expectedWidth, "Got incorrect width: %v", info.Width)
+			assert.Equalf(t, info.Height, tc.expectedHeight, "Got incorrect height: %v", info.Height)
+			assert.Equalf(t, info.HasPreviewImage, tc.expectedHasPreviewImage, "Got incorrect has preview image: %v", info.HasPreviewImage)
```

Comment:
All the above `.Equalf` and `EqualValuesf`  should have their parameters switched as per the [testify documents](https://godoc.org/github.com/stretchr/testify/assert#Equal), the `.Equal` expects the `expected` value first then the `actual` value.

For example
 `assert.Equalf(t,  tc.filename , info.Name, "Got incorrect filename: %v", info.Name)`